src/visitor.rs

```diff
@@ -223,6 +223,41 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
         self.push_str("{");
         self.trim_spaces_after_opening_brace(b, inner_attrs);
 
+        // Try to detect comments that refer to the block, not the first statement in the block.
+        if has_braces && self.config.style_edition() >= StyleEdition::Edition2024 {
+            let block_line_range = self.psess.lookup_line_range(b.span);
+            if block_line_range.lo != block_line_range.hi {
+                // Skipping if a single line block
+                // Make sure there is no code on the first line we have to worry about.
+                // That would also be ambiguous: what should `if { /* comment */ statement;`
+                // even do -- should the comment be attached to the `if` or the `statement`?
+                // The current logic answers this with "statement".
+                let first_line_contains_stmt = if let Some(first_stmt) = b.stmts.first() {
+                    self.psess.lookup_line_range(first_stmt.span).lo == block_line_range.lo
+                } else {
+                    false
+                };
+                if !first_line_contains_stmt {
+                    let first_line_bounds = self.psess.line_bounds(self.last_pos).unwrap();
+                    let first_line_snip = self
+                        .snippet(mk_sp(self.last_pos, first_line_bounds.end))
+                        .trim();
+
+                    if contains_comment(first_line_snip) {
+                        if let Ok(comment) =
+                            rewrite_comment(first_line_snip, false, self.shape(), self.config)
```

Comment:
SO there's no way to adjust the `shape` that we pass to `rewrite_comment` to tell it how much space it actually has available?

How *does* it usually know how much space it has (e.g. to account for indentation)? I would have thought it looks at the cursor position in the current line in the output, which should always be reliable, but that doesn't seem to be the case.